### PR TITLE
switch to prometheus for metrics

### PR DESCRIPTION
**Pull request by lyoshenka**

### CODE REVIEW

**Comment by sayplastic** [comment]
We should use [labels](https://prometheus.io/docs/practices/naming/#labels) for different types of errors. Otherwise looks good to me

File: internal/metrics/metrics.go [hunk 1]
```diff
@@ -57,12 +59,31 @@ func (s *Server) Stop() {
...
+	DirectionUpload   = "upload"   // to reflector
+	DirectionDownload = "download" // from reflector
+
+	errConnReset        = "conn_reset"
+	errReadConnReset    = "read_conn_reset"
+	errWriteConnReset   = "write_conn_reset"
+	errReadConnTimedOut = "read_conn_timed_out"
+	errWriteBrokenPipe  = "write_broken_pipe"
+	errIOTimeout        = "io_timeout"
+	errUnexpectedEOF    = "unexpected_eof"
```
Comment (sayplastic) [inline internal/metrics/metrics.go]: err prefix is usually reserved for error objects.

Comment (lyoshenka) [reply internal/metrics/metrics.go]: agreed, but i don't have a better pattern. what would you name them?

Comment (sayplastic) [reply internal/metrics/metrics.go]: I would name them for what they are: `labelErrUnexpectedEof`

File: internal/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+func (s *Server) Shutdown() {
+	s.srv.Shutdown(context.Background())
+	s.stop.StopAndWait()
+}
+
+const (
+	ns = "reflector"
+
+	labelDirection = "direction"
+	labelErrorType = "error_type"
```
Comment (sayplastic) [inline internal/metrics/metrics.go]: ...and those are technically label names, so `labelNameErrorType`

**Pull request merged**